lib/cpp_peripheral_states.dart

```diff
@@ -0,0 +1,445 @@
+  @override
+  Future<void> handlePeripheralCommand(String cmd) async {
+    if (cmd.startsWith(Command.state)) {
+      round.isMoveRejected = false;
+      round.fen = getCommandParams(cmd);
+      sendRoundUpdateToCentral();
+    } else if (cmd.startsWith(Command.err)) {
+      sendErrToCentral(getCommandParams(cmd));
+    } else if (cmd.startsWith(Command.msg)) {
+      sendMsgToCentral(getCommandParams(cmd));
+    } else if (cmd.startsWith(Command.setOption)) {
+      if (!context.cppOptions.set(getCommandParams(cmd))) {
+        sendErrToCentral('Option set failed: $cmd');
+      } else if (context.areCppOptionsInitialized) {
+        sendOptionsUpdateToCentral();
+      }
+    } else if (cmd.startsWith(Command.optionsEnd)) {
+      context.areCppOptionsInitialized = true;
+      sendOptionsUpdateToCentral();
+    } else if (cmd.startsWith(Command.optionsReset)) {
+      context.cppOptions.reset();
+      sendOptionsUpdateToCentral();
+    } else if (cmd.startsWith(Command.option)) {
+      if (!context.cppOptions.add(getCommandParams(cmd))) {
+        sendErrToCentral('Option add failed: $cmd');
+      }
+    } else {
+      await super.handlePeripheralCommand(cmd);
+    }
+  }
+
+  @override
+  Future<void> handleCentralBegin({
+    required String fen,
+    String? variant,
+    String? side,
+    String? lastMove,
+    String? check,
+    String? time,
+  }) async {
+    round.lastMove = null;
+    round.isMoveRejected = false;
+
+    if (variant != null && round.variant != variant) {
+      await sendCommandToPrtipheral(join(Command.setVariant, variant));
+      round.isVariantSupported = isVariantSupported(variant);
+      if (!round.isVariantSupported) {
+        round.variant = variant;
+        transitionTo(IdleState());
+        sendRoundInitializedToCentral();
+        return;
+      }
+    }
+    round.variant = variant;
+
+    if (side != null && isFeatureSupported(Feature.side)) {
+      await sendCommandToPrtipheral(join(Command.side, side));
+    }
+
+    transitionTo(RoundBeginState());
+    await sendCommandToPrtipheral(join(Command.begin, fen));
+
+    if (lastMove != null && isFeatureSupported(Feature.lastMove)) {
+      await sendCommandToPrtipheral(join(Command.lastMove, lastMove));
+    }
+    if (check != null && isFeatureSupported(Feature.check)) {
+      await sendCommandToPrtipheral(join(Command.check, check));
+    }
+    if (time != null && isFeatureSupported(Feature.time)) {
+      await sendCommandToPrtipheral(join(Command.time, time));
+    }
+  }
+
+  @override
+  Future<void> handleCentralErr({
+    required String err,
+  }) async {
+    await sendCommandToPrtipheral(join(Command.err, err));
+  }
+
+  @override
+  Future<void> handleCentralMsg({
+    required String msg,
+  }) async {
+    if (isFeatureSupported(Feature.msg)) {
+      await sendCommandToPrtipheral(join(Command.msg, msg));
+    }
+  }
+
+  @override
+  Future<void> handleCentralGetState() async {
```

Comment:
shall we have mechanism which will protect central app  to call something what is not supported in peripheral (feature is not set)?

Reply:
I believe it should be an library client role, to check if feature is supported and display `Start from board state` button that trigger `handleCentralGetState`

Reply:
Added to README